api/include/opentelemetry/context/runtime_context.h

```diff
@@ -125,5 +194,116 @@ inline Token::ContextDetacher::~ContextDetacher()
   token.context_ = context_;
   context::RuntimeContext::Detach(token);
 }
+
+// The ThreadLocalContextStorage class is a derived class from RuntimeContext and
+// provides a wrapper for propogating context through cpp thread locally.
+// This file must be included to use the RuntimeContext class if another
+// implementation has not been registered.
+class ThreadLocalContextStorage : public RuntimeContextStorage
+{
+public:
+  ThreadLocalContextStorage() noexcept = default;
+
+  // Return the current context.
+  Context GetCurrent() noexcept override { return GetStack().Top(); }
+
+  // Resets the context to a previous value stored in the
+  // passed in token. Returns true if successful, false otherwise
+  bool Detach(Token &token) noexcept override
+  {
+    if (!(token == GetStack().Top()))
+    {
+      return false;
+    }
+    GetStack().Pop();
+    return true;
+  }
+
+  // Sets the current 'Context' object. Returns a token
+  // that can be used to reset to the previous Context.
+  Token Attach(Context context) noexcept override
+  {
+    GetStack().Push(context);
+    Token old_context = CreateToken(context);
+    return old_context;
+  }
+
+private:
+  // A nested class to store the attached contexts in a stack.
+  class Stack
+  {
+    friend class ThreadLocalContextStorage;
+
+    Stack() noexcept : size_(0), capacity_(0), base_(nullptr){};
+
+    // Pops the top Context off the stack and returns it.
+    Context Pop() noexcept
+    {
+      if (size_ <= 0)
```

Comment:
`size_t` is `unsigned`, so just check `size_ == 0` here?

Reply:
Done.